keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/PartialDerivatives.java

```diff
@@ -105,27 +106,60 @@ public void putWithRespectTo(VertexId id, DoubleTensor value) {
 
     /**
      * This will sum partial derivatives that are represented as tensors over given dimensions.
-     * There is the option to reshape to a lower rank tensor where the summation has caused a
-     * dimension to go to length 1.
      *
-     * @param reshape        Returns the sum and drops the summed over dimensions (now length one)
-     *                       in the shape if true. Returns a same ranked tensor but with a shape
-     *                       that has ones for the dimensions summed over.
-     * @param overDimensions The dimensions to sum over. Dimensions are counted from zero
+     * @param summingAllOfDimensions If True, this means we're summing all of the Of dimensions.
+     *                               In this case, we drop the summed over dimensions from the shape and replace them
+     *                               with a 1x1 (as the Ofs can't disappear completely)
+     * @param overDimensions         The dimensions to sum over. Dimensions are counted from zero
      * @return The summed partial derivatives over given dimensions
      */
-    public PartialDerivatives sum(boolean reshape, int... overDimensions) {
+    public PartialDerivatives sum(boolean summingAllOfDimensions, int... overDimensions) {
```

Comment:
We don't need this anymore right?

Reply:
I decided to leave it in in case we do per dimension summing - although given it will be broken in certain cases for that too - maybe best to get rid of it for now as you say.